timezones/full/build.gradle.kts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2019-2024 JetBrains s.r.o. and contributors.
+ * Use of this source code is governed by the Apache 2.0 License that can be found in the LICENSE.txt file.
+ */
+
+import TzdbTasks.ConvertType
+import TzdbTasks.InstallTimeTzdb
+import TzdbTasks.TzdbDownloadAndCompile
+import org.jetbrains.kotlin.gradle.targets.js.dsl.ExperimentalWasmDsl
+import org.jetbrains.kotlin.gradle.targets.js.npm.NpmResolverPlugin
+
+plugins {
+    kotlin("multiplatform")
```

Comment:
It's a good call to have `timezonesMajorVersion` so we could describe the compatibility between `kotlinx-datetime` and the timezones package. It's easy to imagine people sticking to stable versions of the library but still upgrading the time zones.

But the Kotlin version is also part of the compatibility: there are no compatibility guarantees when someone uses the library compiled with the compiler version newer than the one used by the project itself, so to upgrade the timezone database *correctly*, you will need to upgrade Kotlin, which I think is strange.

I propose this scheme: `kotlinx-datetime-timezones-full` is compiled with the oldest compiler that works, and when some incompatibility arises between the compiler used for `kotlinx-datetime` and the one for `kotlinx-datetime-timezones-full`, only then to upgrade the compiler here, while simultaneously upgrading the `timezonesMajorVersion`.

Thoughts?

Reply:
The only thing here is that we have no possibility (in current approach) to stick a range of major versions of `timezones` to specific range of `datetime` versions in gradle. So for now it could be only documented as compatibility table.